api/tests/src/db/models/factories.py

```diff
@@ -234,6 +236,27 @@ class Meta:
         sqlalchemy_session_persistence = "commit"
 
 
+class OpportunityAttachmentFactory(BaseFactory):
```

Comment:
Could you have the opportunity factory automatically add attachments for itself?

Should be something like:
```py
    opportunity_attachments = factory.RelatedFactoryList(
        OpportunityAttachmentFactory,
        factory_related_name="opportunity",
        size=lambda: random.randint(1, 2),
    )
```

Reply:
Yep, I played around with that earlier but was getting some SQLAlchemy errors, but this snippet works. Thanks

Reply:
Yeah, it's very precise what you need to specify for these related factories to avoid SQLAlchemy or circular dependency issues - but they sure make things easier once they work.

Reply:
Actually, I spoke too soon. The problem seems to be transient -- tests work once, then fail again with an error like:

```
sqlalchemy.exc.InvalidRequestError: Instance <Opportunity at 0xffff81e4c350> is already associated with an instance of <class 'src.db.models.opportunity_models.OpportunityAttachment'> via its OpportunityAttachment.opportunity attribute, and is only allowed a single parent. (Background on this error at: https://sqlalche.me/e/20/bbf1)
```

Reply:
Let me take a look

Reply:
Hmm, it's something with the `single_parent=True` in the opportunity relationship. The pattern here of the tables is effectively the same as the assistance listing table (an opportunity has multiple assistance listings / attachments), but the assistance listing table doesn't have that flag either.

I'd just switch the relationship on the attachment table to: `opportunity: Mapped[Opportunity] = relationship(Opportunity)` - I'm not sure what issue the single_parent thing is getting hung up on because that would imply it's attempting to have the opportunity changed, but if you generate several opportunities, there isn't anything that exists.

Reply:
Works now, thanks!